File: Mennu/Models/Ingredient.cs
using Mennu.Models.Db;
using System;
using System.Collections.Generic;

namespace Mennu.Models
{
    public partial class Ingredient
    {
        public Ingredient(Ingredients ingredient)
        {
            this.Id = ingredient.Id;
            this.Name = ingredient.Name;
            this.Measurement = ingredient.Measurement;
            this.Amount = ingredient.Amount;
        }

        public string Id { get; set; }
        public string Name { get; set; }
        public int? Measurement { get; set; }
        public int? Amount { get; private set; }
    }
}

File: Mennu/wwwroot/Assets/js/models/ingredient.ts
export default class Ingredient {

    constructor(data: any) {
        if (data) {
            this.id = data.id;
            this.name = data.name;
            this.amount = data.amount;
            this.measurement = data.measurement;
        }
    }

    id:string = ""
    name: string = "";
    amount: number = null;
    measurement: number = null;
}